core/src/html/layout.rs

```diff
@@ -469,16 +459,21 @@ impl<'a, 'gc> LayoutContext<'a, 'gc> {
 
         self.append_underlines();
 
-        line_bounds += Position::from((left_adjustment + align_adjustment, baseline_adjustment));
-        line_bounds += Size::from((Twips::ZERO, font_leading_adjustment));
+        line_size_bounds +=
+            Position::from((left_adjustment + align_adjustment, baseline_adjustment));
 
-        self.flush_line(end);
+        if self.current_line_index == 0 {
+            // The very first line always gets the leading.
+            line_size_bounds += Size::from((Twips::ZERO, self.max_leading));
+        }
 
-        if let Some(eb) = &mut self.exterior_bounds {
-            *eb += line_bounds;
+        if self.boxes.first().unwrap().start() == end && last_line {
```

Comment:
is it possible that `self.boxes.is_empty()` here?

Reply:
It's not possible, at the beginning of the method a box is added in case there are no boxes. I've moved this `unwrap` higher so it's more obvious.

Reply:
Oh, I see. I wasn't familiar with the abstractions and didn't connect that `append_text()` makes  `self.boxes` no longer empty.